viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+
+    new_movie = {}
+
+    # check that all necessary movie characteristics exist before creating new movie
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    else:
+        new_movie = None
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+
+    # add movie to watched list
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    # add movie to watchlist
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, movie_title):
+
+    # move movies that user watches from watchlist to watched
+    for movie in user_data["watchlist"]:
+        if movie["title"] == movie_title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    # variable to hold sum of all movie ratings
+    total = 0
+
+    for movie in user_data["watched"]:
+        # add each movie's rating to the total
+        total += movie["rating"]
+    
+    # calculate average rating for all movies within "watched"
+    if total == 0:
+        rating_average = 0
+    else:
+        rating_average = total/len(user_data["watched"])
+
+    return rating_average
+
+def get_most_watched_genre(user_data):
+    
+    # empty dictionary to hold movie genres and their frequency
+    movie_genre_count = {}
+
+    # check if watched list is empty; if so, there will be no most watched genre
+    if not user_data["watched"]:
+        popular_genre = None
+    else:
+        # update movie_genre_count based on watch history
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            if genre not in movie_genre_count:
+                movie_genre_count[genre] = 1
+            else:
+                movie_genre_count[genre] += 1
+
+        # convert movie_genre_count keys and values to lists to allow for max function
+        key_list = list(movie_genre_count.keys())
+        value_list = list(movie_genre_count.values())
+        max_value = max(value_list)
+
+        # identify index of most popular genre
+        popular_genre_index = value_list.index(max_value)
+
+        # identify the most frequently watched movie genre
+        popular_genre = key_list[popular_genre_index]

```

Comment:
The `max` function can work with dictionaries too! Here's another approach we could take using `max`:
```suggestion
        popular_genre = max(movie_genre_count, key=movie_genre_count.get)
```

Reply:
Oh wow, thank you! In addition to being a cleaner, more succinct approach, I also appreciate that your suggestion would have lower time and space complexity since it doesn't create a new list or rely on the index method.